### DISCUSSION TITLE
Fresh install breaks on driver/adc_deprecated.h

### DISCUSSION THREAD

**Discussion by scoates** [PlummersSoftwareLLC/NightDriverStrip#165] [General]
Hi.

I just did a brand new install of everything (fresh git checkout of NightDriverStrip and fresh VSCode PlatformIO install). On my macOS 12.5 MBP.

I tried to `pio run` from a PlatformIO shell (and `PlatformIO: Build` from VSCode) and kept getting an error on something like `Couldn't find driver/adc_deprecated.h` (I've since fixed this and can no longer seem to replicate the original error).

From the pio shell, I did `pio pkg install`, and that failed on `Unknown board ID 'adafruit_feather_esp32s3_tft'`.

So then I tried `pio pkg update` and that seems to have solved my problem. I can `pio run` now.

Thought that might help someone.

**Comment by ljarvie** [comment]
THANK YOU for posting this.  I have almost no knowledge working in this system and am trying to learn by example.  I was getting the errors listed below if I did a clean pull and tried to build it.  I ran the pio pkg install and update and it builds fine now.  Thanks!

Compiling .pio\build\spectrum\lib774\IRremoteESP8266\IRtext.cpp.o
In file included from src\audio.cpp:36:0:
include/soundanalyzer.h:49:35: fatal error: driver/adc_deprecated.h: No such file or directory
compilation terminated.
*** [.pio\build\spectrum\src\audio.cpp.o] Error 1
In file included from include/effects/strip/particles.h:47:0,
                 from include/effects/matrix/spectrumeffects.h:44,
                 from src\drawing.cpp:40:
include/soundanalyzer.h:49:35: fatal error: driver/adc_deprecated.h: No such file or directory
compilation terminated.
*** [.pio\build\spectrum\src\drawing.cpp.o] Error 1
In file included from include/effects/strip/meteoreffect.h:44:0,
                 from src\effects.cpp:35:
include/soundanalyzer.h:49:35: fatal error: driver/adc_deprecated.h: No such file or directory
compilation terminated.
*** [.pio\build\spectrum\src\effects.cpp.o] Error 1
In file included from src\main.cpp:163:0:
include/soundanalyzer.h:49:35: fatal error: driver/adc_deprecated.h: No such file or directory
compilation terminated.
*** [.pio\build\spectrum\src\main.cpp.o] Error 1
In file included from src\screen.cpp:39:0:
include/soundanalyzer.h:49:35: fatal error: driver/adc_deprecated.h: No such file or directory
compilation terminated.
*** [.pio\build\spectrum\src\screen.cpp.o] Error 1
In file included from include/network.h:31:0,
                 from src\network.cpp:32:
include/socketserver.h:75:34: error: expected ',' before ')' token
 static_assert(sizeof(double) == 8);
                                  ^
include/socketserver.h:75:34: error: expected string-literal before ')' token
In file included from src\network.cpp:34:0:
include/spiffswebserver.h: In member function 'void CSPIFFSWebServer::GetStatistics(AsyncWebServerRequest*)':
include/spiffswebserver.h:151:77: error: 'heap_caps_get_total_size' was not declared in this scope
         j["DMA_SIZE"]              = heap_caps_get_total_size(MALLOC_CAP_DMA);
                                                                             ^
include/spiffswebserver.h:159:42: error: 'class EspClass' has no member named 'getChipModel'
         j["CHIP_MODEL"]            = ESP.getChipModel();
                                          ^
include/spiffswebserver.h:160:42: error: 'class EspClass' has no member named 'getChipCores'
         j["CHIP_CORES"]            = ESP.getChipCores();
                                          ^
include/spiffswebserver.h: In lambda function:
include/spiffswebserver.h:290:69: warning: format '%d' expects argument of type 'int', but argument 3 has type 'fs::File*' [-Wformat=]
                 Serial.printf("[HTTP]\tOpening [%d]\r\n", SPIFFSfile);
                                                                     ^
include/spiffswebserver.h:294:102: warning: format '%d' expects argument of type 'int', but argument 3 has type 'fs::File*' [-Wformat=]
                 Serial.printf("[HTTP]\tSPIFFS File DOESN'T exists [%d] <<< ERROR !!!\r\n", SPIFFSfile);
                                                                                                      ^
*** [.pio\build\spectrum\src\network.cpp.o] Error 1
======================================================================================= [FAILED] Took 10.87 seconds =======================================================================================
Environment    Status    Duration
-------------  --------  ------------
spectrum       FAILED    00:00:10.874
================================================================================== 1 failed, 0 succeeded in 00:00:10.874 =================================================================
